mlflow/data/huggingface_dataset.py

```diff
@@ -110,7 +110,7 @@ def source(self) -> HuggingFaceDatasetSource:
         """
         Hugging Face dataset source information.
 
-        :return: A :py:class:`HuggingFaceDatasetSource` instance.
+        :return: A :py:class:`mlflow.data.huggingface_dataset_source.HuggingFaceDatasetSource` instance.
```

Comment:
Can we replace single backticks in docstrings with double backticks and also add `:py:class` and `:py:func` links where applicable. For example, `dataset.Dataset` should be linked with `:py:class` in the HuggingFace docstring:

<img width="819" alt="Screenshot 2023-05-26 at 3 32 37 PM" src="https://github.com/mlflow/mlflow/assets/39497902/779bb4e1-ed04-45ea-b1d4-41f1fbfff6dc">

Double backticks for literals are preferred because they render as bold text instead of italics.

Can we make these changes across documented classes / methods for the other datasets / dataset sources as well?

Reply:
@dbczumar I've updated this across various pages. 

However, I'm not sure how to fix this particular example. `datasets.Dataset` is a link to external documentation https://huggingface.co/docs/datasets/package_reference/main_classes#datasets.Dataset. Do we have examples of this in our current docs? I haven't seen yet. 

BTW, if using `:py:class:`/`:py:func:`, then I think we need to be using single backticks to show as a link. But if not `:py:class:`/`:py:func:`, then double backticks is needed to show as bold.

Reply:
We may need to use something like https://stackoverflow.com/questions/44682721/how-to-link-to-external-libraries-using-python-sphinx for external references